bgpd/bgp_vty.c

```diff
@@ -11375,15 +11379,19 @@ static int bgp_show_route_leak_vty(struct vty *vty, const char *name,
 				json_object_array_add(json_import_vrfs,
 						json_object_new_string(vname));
 
+			json_object_object_add(json, "importFromVrfs",
+						       json_import_vrfs);
 			dir = BGP_VPN_POLICY_DIR_FROMVPN;
-			ecom_str = ecommunity_ecom2str(
+			if (bgp->vpn_policy[afi].rtlist[dir]) {
+				ecom_str = ecommunity_ecom2str(
 					bgp->vpn_policy[afi].rtlist[dir],
 					ECOMMUNITY_FORMAT_ROUTE_MAP, 0);
-			json_object_object_add(json, "importFromVrfs",
-					       json_import_vrfs);
```

Comment:
new code doesn't add this object, is this correct?

Reply:
It does at line 11382. So first we add the object, then if rtlist contains list of import ecommunity then string importRTs with ecom_str printed otherwise simply print  "none" with it.
Earlier since e communities were present we add the object and print imports. With source bgp instance vanished import rts can be empty.